### ISSUE TITLE
Return richer information about accounts

### ISSUE THREAD

**Issue by GalenReich** [bellingcat/telegram-phone-number-checker#16]
At the moment, the tool makes the `id`, `username`, `first_name`, `last_name` fields available for each account that is successfully found. 

Telethon returns several more fields associated with a User, as [detailed in their api](https://tl.telethon.dev/constructors/user.html).

Some of these fields (`fake`, `scam`, `premium`,`verified`, etc...) look like they could be of interest to researchers. This may need some research, as it is unclear how widely these fields are used and if they are likely to be accurate.

It would be good if useful fields could be identified and returned to the tool user.

**Comment by BDFL669** [comment]
I would really love to work on this issue! Please assign it to me

**Comment by GalenReich** [comment]
Hi @BDFL669 👋 Thanks for reaching out - we'd really welcome a pull request on this issue - I'm happy to assign it to you for now. If you don't have time to work on it, just let me know and I'll unassign you later!

Also, if you're interested in being part of the wider Bellingcat community, you can join our [Bellingcat Discord](https://discord.gg/bellingcat) too.

**Comment by BDFL669** [comment]
I'm happy to help! Lemme get to work

**Comment by GalenReich** [comment]
Looking at the other issues here, this might solve #9 too, though finding a banned number for testing might be difficult.

**Comment by GalenReich** [comment]
Hi @BDFL669, just wanted to check in and see how you were getting on. Have you found anything of interest about the fields?

**Comment by BDFL669** [comment]
hi @GalenReich, sorry for the late response. I've been playing around with the app and most of the entries are not relevant to our usecase. However, I've added new fields I think you might find useful. 
![image](https://github.com/bellingcat/telegram-phone-number-checker/assets/62002130/f4acd4f4-8edc-4c09-8469-904af7f22a2e)

**Comment by GalenReich** [comment]
@BDFL669 said:

> hi @GalenReich, sorry for the late response. I've been playing around with the app and most of the entries are not relevant to our usecase. However, I've added new fields I think you might find useful.
```
Checking: '{Phone Number Removed}' ...Done
{
  "{Phone number removed}": {
    "id": {ID removed},
    "username": "{Username removed}" ,
    "first name": "{Name removed}" ,
    "last name" : null,
    "fake": false,
    "scam": null,
    "verified": false,
    "premium": false,
    "mutual contact": false,
    "user status" : null,
    "bot": false,
    "bot chat history": false,
    "restricted": false,
    "restriction_reason": [ ]
  }
}
```

Great stuff! Just wanted to check in! I'm going to delete your last message so that the image with the phone number and user information is not preserved, but have quoted it above - it looks good👍

**Comment by BDFL669** [comment]
Awesome, thank you so much! Would you like me to submit a PR with the
latest changes?

**Comment by GalenReich** [comment]
Feel free to open a PR or a Draft PR - that would be good - the main thing will be to understand to how common those fields are. For example, does Telegram ever set the `scam` field?

It might be good to include `status.was_online` and have some parsing to make it human readable

**Comment by BDFL669** [comment]
Thanks for the tip!

**Comment by BDFL669** [comment]
Greetings! The API seldom provides information for the scam parameter so
I've had to remove it. Moreover, After parsing status.was_online data, I
assume this covers the user status field so I've also removed this one as
well (telegram rarely propagates this field as well). Here is a screenshot
of my output after redacting any revealing details about my telegram
account. If the results are satisfactory I shall proceed to open a PR

[image: image.png]

**Comment by BDFL669** [comment]
here is the screenshot: 
![image](https://github.com/bellingcat/telegram-phone-number-checker/assets/62002130/3e3c4f39-26ac-49f7-942a-dc9c986b6bab)

**Comment by BDFL669** [comment]
Greetings! The API seldom provides information for the scam parameter so
I've had to remove it. Moreover, After parsing status.was_online data, I
assume this covers the user status field so I've also removed this one as
well (telegram rarely propagates this field as well). Here is a screenshot
of my output after redacting any revealing details about my telegram
account. If the results are satisfactory I shall proceed to open a PR

![image](https://github.com/bellingcat/telegram-phone-number-checker/assets/62002130/9d52dba5-e303-466e-8d95-b37d8baab02f)
